processor/processor.go

```diff
@@ -1482,6 +1501,12 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 		requestIP := gatewayBatchEvent.RequestIP
 		receivedAt := gatewayBatchEvent.ReceivedAt
 
+		// Only if geo enrichment is enabled for deployment and then the source
+		// only then we perform the enrichment on gateway batch event.
+		if proc.config.geoEnrichmentEnabled && proc.geoEnrichmentEnabledForSource(sourceId) {
+			_ = proc.enricher.Enrich(sourceId, &gatewayBatchEvent)
+		}
+
```

Comment:
- If geoenrichment is disabled then no need to add the geoenricher in processor.
- The geoenricher itself should know which sources have geoenrichment enabled or not, processor shouldn't know anything about geoenricher, only have a list of enrichers.
```suggestion
		for _, enricher := range proc.enrichers {
			_ = proc.enricher.Enrich(sourceId, &gatewayBatchEvent)
		}
```